lib/LedgerSMB/X12.pm

```diff
@@ -112,7 +112,7 @@ sub _ISA {
 
     my @keys;
 
-    push @keys, sprintf('ISA%02d', $_) for (1 .. 16);
+    push @keys, sprintf('ISA%02d', $_) for (1 .. 16);  ## no critic (ProhibitMagicNumbers) sniff
```

Comment:
Referring to http://edicrossroad.blogspot.nl/2008/12/isa-and-gs-segment-elements-enumeration.html , 1 is the minimum ISA header number and 16 the max. So, it seems we can safely create constants for these.

Reply:
I remember not finding any authoritative naming for the EDI fields.  I cannot read this page now.  Do you accept it as authoritative?  Does it have more than "ISA07" for names?

I suppose the loop could be unrolled. 

What do you think?

Reply:
Yes, the page has a mapping between codes and description:

```
    ISA01 Authorization Information Qualifier : min/max – 2/2
    ISA02 Authorization Information : min/max – 10/10
    ISA03 Security Information Qualifier : min/max – 2/2
    ISA04 Security Information : min/max – 10/10
    ISA05 Interchange ID Qualifier : min/max – 2/2
    ISA06 Interchange Sender ID : min/max – 15/15
    ISA07 Interchange ID Qualifier : min/max – 2/2
    ISA08 Interchange Receiver ID : min/max – 15/15
    ISA09 Interchange Date : min/max – 6/6
    ISA10 Interchange Time : min/max – 4/4
    ISA11 Interchange Control Standards ID : min/max – 1/1
    ISA12 Interchange Control Version Number : min/max – 5/5
    ISA13 Interchange Control Number : min/max – 9/9
    ISA14 Acknowledgment Requested : min/max – 1/1
    ISA15 Test Indicator : min/max – 1/1
    ISA16 Subelement Separator : min/max – 1/1
```

With this information, unrolling the loop is probably the best option we have.